### PR TITLE
PYTHON-3061 Add 'let' option to ReplaceOptions

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

**Review by blink1073 — APPROVED** [review]
LGTM!

File: pymongo/collection.py [hunk 1]
```diff
@@ -673,7 +673,7 @@ def _update(session, sock_info, retryable_write):
 
     def replace_one(self, filter, replacement, upsert=False,
                     bypass_document_validation=False, collation=None,
-                    hint=None, session=None):
+                    hint=None, session=None, let=None):
```
Comment (ShaneHarvey) [inline pymongo/collection.py:676]: Can you open a new ticket to add `let` to the bulk_write models (UpdateOne, ReplaceOne, etc)? We neglected to do that in PYTHON-2957.

File: pymongo/collection.py [hunk 2]
```diff
@@ -721,10 +721,16 @@ def replace_one(self, filter, replacement, upsert=False,
...
+          - `let` (optional): Map of parameter names and values. Values must be
+            constant or closed expressions that do not reference document
+            fields. Parameters can then be accessed as variables in an
+            aggregate expression context (e.g. "$$var").
 
         :Returns:
           - An instance of :class:`~pymongo.results.UpdateResult`.
 
+        .. versionchanged:: 4.1
+           Added ``let`` parameter.
```
Comment (ShaneHarvey) [inline pymongo/collection.py:733]: Please also update the changelog.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
LGTM so far but still need a changelog.

File: pymongo/collection.py (hunk 1 above)
Comment (juliusgeo) [reply pymongo/collection.py:676]: https://jira.mongodb.org/browse/PYTHON-3068

File: pymongo/collection.py (hunk 2 above)
Comment (juliusgeo) [reply pymongo/collection.py:733]: Done.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**